detections/application/splunk_command_and_scripting_interpreter_detect_risky_spl_mltk.yml

```diff
@@ -0,0 +1,58 @@
+name: Splunk Command and Scripting Interpreter Detect Risky SPL MLTK
+id: b4aefb5f-1037-410d-a149-1e091288ba33
+version: 1
+date: '2022-06-16'
+author: Abhinav Mishra, Kumar Sharad, Namratha Sreekanta and Xiao Lin, Splunk
+type: Anomaly
+datamodel:
+- Splunk_Audit
+description: 'The following analytic uses a pre-trained machine learning text classifier to detect potentially risky commands. A command is deemed risky based on the presence of certain trigger keywords, along with the context and the role of the user (please see references). The model takes as input the command text, user and search type and outputs a risk score between [0,1]. A high score indicates higher likelihood of a command being risky.'
+search: '| tstats `security_content_summariesonly` count min(_time) as firstTime max(_time) as lastTime from datamodel=Splunk_Audit.Search_Activity by Search_Activity.search Search_Activity.info Search_Activity.total_run_time Search_Activity.user Search_Activity.savedsearch_name Search_Activity.search_type
+  | eval spl_text = 'Search_Activity.search'. " " .'Search_Activity.user'. " " .'Search_Activity.search_type'
+  | dedup spl_text
+  | apply DetectRiskySPL
+  | table Search_Activity.search, Search_Activity.user, Search_Activity.search_type, risk_score
```

Comment:
There is no filtering based on risk score, so a risk event will be created for every event, which isn't what we want here.

Reply:
Done!